modules/auxiliary/scanner/http/dell_idrac.rb

```diff
@@ -81,7 +92,43 @@ def do_login(user = nil, pass = nil)
       )
       return :next_user
     else
-      print_error("#{target_url} - Dell iDRAC - Failed to login as '#{user}' with password '#{pass}'")
+      vprint_error("#{target_url} - Dell iDRAC - Failed to login as '#{user}' with password '#{pass}'")
+    end
+  end
+
+  def do_login_v9(user = nil, pass = nil)
+    uri = v9_url
+    auth = send_request_cgi({
+      'method' => 'POST',
+      'uri' => uri,
+      'SSL' => true,
+      'headers' => { 'user' => user, 'password' => pass },
+      'vars_post' => {
+        'user' => user,
+        'password' => pass
+      }
+    })
+    unless auth
+      print_error('iDRAC failed to respond to login attempt')
+      return :next_user # assume this is a temporary error
+    end
+    json = JSON.parse(auth.body)
```

Comment:
Do we need to detect the scenario where we don't get back json, that seems like we should abort in that scenario?

In that scenario I believe `JSON.parse` would also throw an exception, potentially we should use `res.get_json_document`:

```suggestion
    json = JSON.parse(auth.body)
```


which would give us back `nil` in the error scenario:

https://github.com/rapid7/metasploit-framework/blob/9b76ecf6c7c7436cb0a4bb83395ddc1bb79bc38a/lib/rex/proto/http/response.rb#L119-L133